diff --git a/litex_boards/platforms/digilent_basys3.py b/litex_boards/platforms/digilent_basys3.py
index d466109..cbbf643 100644
--- a/litex_boards/platforms/digilent_basys3.py
+++ b/litex_boards/platforms/digilent_basys3.py
@@ -133,7 +133,7 @@ class Platform(Xilinx7SeriesPlatform):
     default_clk_period = 1e9/100e6
 
     def __init__(self, toolchain="vivado"):
-        Xilinx7SeriesPlatform.__init__(self, "xc7a35t-CPG236-1", _io, _connectors, toolchain=toolchain)
+        Xilinx7SeriesPlatform.__init__(self, "xc7a35tcpg236-1", _io, _connectors, toolchain=toolchain)
 
     def create_programmer(self):
         return OpenOCD("openocd_xc7_ft2232.cfg", "bscan_spi_xc7a35t.bit")
diff --git a/litex_boards/targets/digilent_basys3.py b/litex_boards/targets/digilent_basys3.py
index ae5e1e7..a70d019 100755
--- a/litex_boards/targets/digilent_basys3.py
+++ b/litex_boards/targets/digilent_basys3.py
@@ -39,8 +39,8 @@ class _CRG(LiteXModule):
 # BaseSoC ------------------------------------------------------------------------------------------
 
 class BaseSoC(SoCCore):
-    def __init__(self, sys_clk_freq=75e6, with_led_chaser=True, with_video_terminal=False, **kwargs):
-        platform = digilent_basys3.Platform()
+    def __init__(self, toolchain="vivado", sys_clk_freq=75e6, with_led_chaser=True, with_video_terminal=False, **kwargs):
+        platform = digilent_basys3.Platform(toolchain=toolchain)
 
         # CRG --------------------------------------------------------------------------------------
         self.crg = _CRG(platform, sys_clk_freq)
@@ -74,6 +74,7 @@ def main():
     args = parser.parse_args()
 
     soc = BaseSoC(
+        toolchain           = args.toolchain,
         sys_clk_freq        = args.sys_clk_freq,
         with_video_terminal = args.with_video_terminal,
         **parser.soc_argdict
