diff --git a/vpr/src/base/old_traceback.cpp b/vpr/src/base/old_traceback.cpp
index 317a4fe27..664e45b99 100644
--- a/vpr/src/base/old_traceback.cpp
+++ b/vpr/src/base/old_traceback.cpp
@@ -44,7 +44,7 @@ void TracebackCompat::traceback_to_route_tree_x(t_trace* trace, RouteTree& tree,
     else
         new_node->re_expand = true;
 
-    if (rr_graph.node_type(inode) == SINK) {
+    if (trace->iswitch == OPEN) {
         /* The traceback returns to the previous branch point if there is more than one SINK, otherwise we are at the last SINK */
         if (trace->next) {
             RRNodeId next_rr_node = RRNodeId(trace->next->index);
