diff --git a/f4pga/wrappers/tcl/xc7.f4pga.tcl b/f4pga/wrappers/tcl/xc7.f4pga.tcl
index dbab425..1e00a01 100644
--- a/f4pga/wrappers/tcl/xc7.f4pga.tcl
+++ b/f4pga/wrappers/tcl/xc7.f4pga.tcl
@@ -208,14 +208,14 @@ if { $::env(USE_ROI) != "TRUE" } {
 # Re-run optimization flow to absorb carry modifications
 hierarchy -check
 
-write_ilang $::env(OUT_JSON).pre_abc9.ilang
+write_rtlil $::env(OUT_JSON).pre_abc9.ilang
 if { $::env(USE_ROI) == "TRUE" } {
     synth_xilinx -flatten -abc9 -nosrl -noclkbuf -nodsp -noiopad -nowidelut -run map_ffs:check
 } else {
     synth_xilinx -flatten -abc9 -nosrl -noclkbuf -nodsp -iopad -nowidelut -run map_ffs:check
 }
 
-write_ilang $::env(OUT_JSON).post_abc9.ilang
+write_rtlil $::env(OUT_JSON).post_abc9.ilang
 
 # Either the JSON bounce or ABC9 pass causes the CARRY4_VPR CIN/CYINIT pins
 # to have 0's when unused.  As a result VPR will attempt to route a 0 to those
