--- a/Vivado/@version@/bin/hw_server
+++ b/Vivado/@version@/bin/hw_server
@@ -69,12 +69,7 @@ if [[ "`uname`" == MINGW*NT* ]]; then
   exit $?
 fi
 
-RDI_OS_ARCH=32
-case `uname -m` in
-  x86_64) 
-    RDI_OS_ARCH=64
-    ;;
-esac
+RDI_OS_ARCH=64
 
 #
 # If True check for the existence of RDI_PROG prior to invoking the
--- a/Vivado/@version@/bin/loader
+++ b/Vivado/@version@/bin/loader
@@ -65,15 +65,6 @@ if [[ "`uname`" == MSYS*NT* ]]; then
   exit $?
 fi
 
-case `uname -m` in
-  x86_64)
-    ;;
-  *)
-    echo 1>&2 "Unsupported architecture: `uname -m`"
-    exit 1
-    ;;
-esac
-
 #
 # If True check for the existence of RDI_PROG prior to invoking the
 # Function RDI_EXEC_COMMANDS
--- a/Vivado/@version@/bin/svf_utility
+++ b/Vivado/@version@/bin/svf_utility
@@ -39,11 +39,6 @@ RDI_PROG=`basename "$0"`
 RDI_NO_JRE=yes
 export RDI_NO_JRE
 #######################################
-HW_SERVER_PLATFORM=lnx32
-case `uname -m` in
-  x86_64) 
-    HW_SERVER_PLATFORM=lnx64
-    ;;
-esac
+HW_SERVER_PLATFORM=lnx64
 #######################################
 "$RDI_BINROOT"/loader -exec $RDI_PROG "$@"
--- a/Vivado/@version@/bin/xrt_server
+++ b/Vivado/@version@/bin/xrt_server
@@ -69,12 +69,7 @@ if [[ "`uname`" == MINGW*NT* ]]; then
   exit $?
 fi
 
-RDI_OS_ARCH=32
-case `uname -m` in
-  x86_64) 
-    RDI_OS_ARCH=64
-    ;;
-esac
+RDI_OS_ARCH=64
 
 #
 # If True check for the existence of RDI_PROG prior to invoking the
--- a/Vivado/@version@/bin/xsdb
+++ b/Vivado/@version@/bin/xsdb
@@ -35,12 +35,7 @@
 RDI_NO_JRE=yes
 export RDI_NO_JRE
 
-RDI_PLATFORM=lnx32
-case `uname -m` in
-  x86_64) 
-    RDI_PLATFORM=lnx64
-    ;;
-esac
+RDI_PLATFORM=lnx64
 
 LOG_FILE="$HOME"/.xsdbcmdhistory
 RDI_OPT_EXT=.o
--- a/Vivado/@version@/bin/xtclsh
+++ b/Vivado/@version@/bin/xtclsh
@@ -42,12 +42,7 @@ fi
 RDI_NO_JRE=yes
 export RDI_NO_JRE
 
-PLATFORM_TMP=lnx32
-case `uname -m` in
-  x86_64)
-    PLATFORM_TMP=lnx64
-    ;;
-esac
+PLATFORM_TMP=lnx64
 
 RDI_OPT_EXT=.o
 argSize=0
